app/controllers/staff/sponsors_controller.rb

```diff
@@ -0,0 +1,42 @@
+class Staff::SponsorsController < Staff::ApplicationController
+  before_action :enable_website_subnav
+
+  def index
+    @sponsors = current_event.sponsors
+  end
+
+  def new
+    @sponsor = current_event.sponsors.build
+  end
+
+  def create
+    @sponsor = current_event.sponsors.build(sponsor_params)
+
+    flash[:success] = "Sponsor was successfully created."
+    redirect_to event_staff_sponsors_path if @sponsor.save
```

Comment:
I think I would tend to put the `@sponsor.save` above the flash message like `update`. Ultimately we probably should have some validation like uniqueness on name scoped to the event and this would have more of a typical look of: 
```
  def create
    @sponsor = current_event.sponsors.build(sponsor_params)
    if @sponsor.save
      flash[:success] = "Sponsor was successfully created."
      redirect_to event_staff_sponsors_path if @sponsor.save
    else
       render :new
     end
   end
```